src/main/java/com/fouristhenumber/utilitiesinexcess/common/items/ItemWateringCan.java

```diff
@@ -0,0 +1,393 @@
+                }
+
+                world.setBlock(bx, by, bz, flowerBlock, meta, 2);
+            }
+        }
+    }
+
+    /**
+     * Attempts to multiply a flower near the given coordinates.
+     */
+    private void multiplyFlower(World world, int x, int y, int z, BlockFlower flowerBlock, int flowerMeta) {
+        Block below = world.getBlock(x, y - 1, z);
+
+        if (world.isAirBlock(x, y, z) && below.canSustainPlant(world, x, y - 1, z, ForgeDirection.UP, flowerBlock)) {
+            world.setBlock(x, y, z, flowerBlock, flowerMeta, 2);
+        }
+    }
+
+    /**
+     * Spawns particles around the watering can to indicate its use.
+     * This method is called when the watering can is used to water crops.
+     * It sends a particle packet to all players around the watering can.
+     */
+    public void spawnParticle(World world, int x, int y, int z) {
+        Random rand = world.rand;
+        for (int dx = -range; dx <= range; dx++) {
+            for (int dz = -range; dz <= range; dz++) {
+                double d0 = x + dx + rand.nextFloat();
+                double d1 = y + world.getBlock(x, y, z)
+                    .getBlockBoundsMaxY();
+                double d2 = z + dz + rand.nextFloat();
+
+                if (world.getBlock(x, y, z)
+                    .isOpaqueCube() || world.getBlock(x, y, z) instanceof BlockFarmland) {
+                    d1 += 0.3D;
+                }
+
+                // Send particle packet to all players around the watering can
+                if (!world.isRemote) {
+                    PacketHandler.INSTANCE.sendToAllAround(
+                        new ParticlePacket("splash", d0, d1, d2, 5),
+                        new NetworkRegistry.TargetPoint(world.provider.dimensionId, d0, d1, d2, 32.0 // range in blocks
+                        ));
+                }
+            }
+        }
+    }
+
+    @Override
+    @SideOnly(Side.CLIENT)
+    public void addInformation(ItemStack stack, EntityPlayer player, List<String> tooltip, boolean advanced) {
+        int range = (this.range * 2 + 1);
+        tooltip.add(EnumChatFormatting.AQUA + StatCollector.translateToLocal("item.watering_can.desc.1"));
+        tooltip.add(
+            EnumChatFormatting.AQUA
+                + StatCollector.translateToLocalFormatted("item.watering_can.desc.2", range, range));
+        if (wateringCan.allowWateringCanToggle) {
+            tooltip.add(
+                EnumChatFormatting.YELLOW + EnumChatFormatting.ITALIC.toString()
+                    + StatCollector.translateToLocal("item.watering_can.desc.3"));
+            tooltip.add(
+                EnumChatFormatting.YELLOW + EnumChatFormatting.ITALIC.toString()
+                    + StatCollector.translateToLocal("item.watering_can.desc.4"));
+        }
+    }
+
+    public boolean isActive(ItemStack stack) {
+        NBTTagCompound tag = stack.getTagCompound();
+        return tag != null && tag.getBoolean(TAG_ACTIVE);
```

Comment:
can use stack.hasTagCompound() which just does the tag != null check for you.
```suggestion
return stack.hasTagCompound() && stack.getTagCompound().getBoolean(TAG_ACTIVE);
```